android/src/main/kotlin/com/simform/audio_waveforms/WaveformExtractor.kt

```
@@ -0,0 +1,245 @@
+                            16
+                        }
+                        totalSamples = sampleRate.toLong() * duration
+                        perSamplePoints = totalSamples / expectedPoints
+                    }
+
+                    override fun onError(codec: MediaCodec, e: MediaCodec.CodecException) {
+                        result.error(
+                            Constants.LOG_TAG,
+                            e.message,
+                            "An error is thrown while decoding the audio file"
+                        )
+                        finishCount.countDown()
+                    }
+
+                    override fun onOutputBufferAvailable(
+                        codec: MediaCodec,
+                        index: Int,
+                        info: MediaCodec.BufferInfo
+                    ) {
+                        if (info.size > 0) {
+                            codec.getOutputBuffer(index)?.let { buf ->
+                                val size = info.size
+                                buf.position(info.offset)
+                                when (pcmEncodingBit) {
+                                    8 -> {
+                                        handle8bit(size, buf)
+                                    }
+                                    16 -> {
+                                        handle16bit(size, buf)
+                                    }
+                                    32 -> {
+                                        handle32bit(size, buf)
+                                    }
+                                }
+                                codec.releaseOutputBuffer(index, false)
+                            }
+                        }
+
+                        if (info.isEof()) {
+                            stop()
+                        }
+                    }
+
+                })
+                it.start()
+            }
+
+        } catch (e: Exception) {
```

Do not add empty catch block. Add debug logs.